tests/phpunit/tests/filesystem/wpMkdirP.php

```diff
@@ -0,0 +1,196 @@
+
+		$target = self::$test_directory . $hook_name;
+
+		// Force a failure.
+		$this->touch( $target );
+
+		wp_mkdir_p( $target );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire an action when the target directory contains '../'.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_not_fire_action_when_the_target_contains_path_traversal( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		// Force a failure by not using `realpath()`.
+		$target = DIR_TESTDATA . "/test_wp_mkdir_p/$hook_name/";
+
+		wp_mkdir_p( $target );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire an action when the target directory contains '..DIRECTORY_SEPARATOR'.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_not_fire_action_when_the_target_contains_path_traversal_with_directory_separator( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		// Force a failure.
+		$target = str_replace( '../', '..' . DIRECTORY_SEPARATOR, DIR_TESTDATA ) . "/test_wp_mkdir_p/$hook_name";
+
+		wp_mkdir_p( $target );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @throws Exception
+	 *
+	 * @return array[]
+	 */
+	public function data_actions() {
+		return self::text_array_to_dataprovider( array( 'before_create_directory', 'after_create_directory' ) );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire the 'after_create_directory' action when `mkdir()` fails.
+	 *
+	 * @ticket 44083
+	 */
+	public function test_wp_mkdir_p_should_not_fire_the_after_create_directory_action_when_mkdir_fails() {
+		$action = new MockAction();
+		add_action( 'after_create_directory', array( $action, 'action' ) );
+
+		add_action(
+			'before_create_directory',
+			function( $target ) {
+				/*
+				 * Force a failure by creating a file of the same name
+				 * just before `mkdir()` runs.
+				 */
+				$this->touch( $target );
+			}
+		);
+
+		wp_mkdir_p( self::$test_directory . 'after_create_directory' );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` fires the 'create_directory_failed' action when `mkdir()` fails.
+	 *
+	 * @ticket 44083
+	 */
+	public function test_wp_mkdir_p_should_fire_the_create_directory_failed_action_when_mkdir_fails() {
+		$action = new MockAction();
+		add_action( 'create_directory_failed', array( $action, 'action' ) );
+
+		add_action(
+			'before_create_directory',
+			function( $target ) {
+				/*
+					* Force a failure by creating a file of the same name
+					* just before `mkdir()` runs.
+					*/
```

Comment:
```suggestion
				/*
				 * Force a failure by creating a file of the same name
				 * just before `mkdir()` runs.
				 */
```
Just a minor alignment fix.